### PR TITLE
maint: Cleanup CMake files and delete not compiled files

### PR DESCRIPTION
**Pull request by mathbunnyru**
I did the following:
- sorted the order files are mentioned in cmake files
- added missing entries
- deleted files which are not compiled at all (I guess we don't need them?)

### CODE REVIEW

File: libmamba/CMakeLists.txt
Comment (mathbunnyru) [inline libmamba/CMakeLists.txt:166]: This is the correct sort order, and this is how it shows up in the tree list in VS Code, for example

File: libmamba/CMakeLists.txt
Comment (mathbunnyru) [inline libmamba/CMakeLists.txt:324]: This is an example of missing line. Everything will work except that VS Code will give errors when you open this file - it annoyed me, so I decided to create this PR

File: libmamba/include/mamba/core/mirror.hpp
Comment (mathbunnyru) [inline libmamba/include/mamba/core/mirror.hpp:1]: This file is not included by anyone, and both it and .cpp file are not mentioned in CMake, so I think it's safe to remove it.
Please, make sure I'm right

File: micromamba/src/microserver.cpp
Comment (mathbunnyru) [inline micromamba/src/microserver.cpp:1]: This file and `server.cpp` are not mentioned in CMake at all, so I think it's safe to remove it.
Please, make sure I'm right.

File: libmamba/ext/solv-cpp/tests/CMakeLists.txt
Comment (mathbunnyru) [inline libmamba/ext/solv-cpp/tests/CMakeLists.txt:14]: There is no separation for source and header files here, so we should mention this header file as well.

Comment (mathbunnyru) [reply libmamba/ext/solv-cpp/tests/CMakeLists.txt:14]: Yes.
And it also makes it consistent (all source/header files are present in CMake files).

**Comment by mathbunnyru** [comment]
> LGTM.
> 
> We need to add code coverage to see any potential dead code which may exist, see #2866.
> 
> I just have one minor comment.

Thanks! Replied to your comment.

I agree that having coverage would be nice, but I don't think it would help here.
As far as I know, it only works for source/header files that are actually compiled.
So, if you have a header file, which is not included anywhere, or source file, which is not compiled, it won't not work.
(I might be wrong here)

**Comment by jjerphan** [comment]
I agree with you and I meant that adding code coverage is an addiction to dead code elimination. :slightly_smiling_face:

**Comment by mathbunnyru** [comment]
@jjerphan I think this is ready to be merged as well 🙂

**Comment by jjerphan** [comment]
For this one, I would wait for another review of the team.

File: micromamba/src/server.cpp
Comment (JohanMabille) [inline micromamba/src/server.cpp:1]: This file is still used in [micromamba's CMakeLists.txt](https://github.com/mamba-org/mamba/blob/main/micromamba/CMakeLists.txt#L48-L51), and it includes microserver.cpp, so I don't think we should remove them for now.

**Review by JohanMabille — CHANGES_REQUESTED** [review]

File: micromamba/src/server.cpp
Comment (mathbunnyru) [reply micromamba/src/server.cpp:1]: Just as a note: these files are not compiled in any way right now, and their compilation is significantly broken.

Comment (mathbunnyru) [reply micromamba/src/server.cpp:1]: So, I think, if we want to keep these files, we need to fix the compilation and add it to the CI.
Otherwise we'll just have broken files in the repo, which no one is able to use easily.

Comment (mathbunnyru) [reply micromamba/src/server.cpp:1]: Created an issue: https://github.com/mamba-org/mamba/issues/3677

Comment (JohanMabille) [reply micromamba/src/server.cpp:1]: @mathbunnyru I agree; IIRC we initially wanted to remove these files and then realized we need them for another project (I think Quetz). We kept them for reference because we need to refactor them (ok, maybe it will be a full rewrite, but we still need to keep them for now), but did not have the bandwidth to do it. We'll probably tackle this in 2025.

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
Thanks for the cleanup!

**Pull request merged**
